snapcraft/internal/elf.py

```diff
@@ -56,41 +72,41 @@ def __getitem__(self, key):
         return self._soname_paths[key]
 
     def __setitem__(self, key, item):
```

Comment:
We've probably moved beyond using `__setitem__` now, we could make our API more explicit.